examples/todolist/src/Server.hs

```diff
@@ -67,19 +69,17 @@ list_ t = do
     pure $ TodoListList todos
 
 update :: TodoList -> TodoListMessage -> IO TodoListResponse
-update t mg@(TodoListMessage idM titM tgM) = do
-  putStr "update: " >> print (idM, titM, tgM)
-  atomically $ do
-    modifyTVar t (fmap (\m -> if getMsg idM m then mg else m))
-    pure $ TodoListResponse mg
+update t mg@(TodoListMessage idM titM tgM compl) = do
+  putStr "update: " >> print (idM, titM, tgM, compl)
+  atomically $ modifyTVar t (fmap (\m -> if getMsg idM m then mg else m))
+  pure $ TodoListResponse mg
 
 destroy :: TodoList -> MessageId -> IO MessageId
 destroy t (MessageId idMsg) = do
   putStr "destroy: " >> print idMsg
-  atomically $ do
-    todos <- readTVar t
-    case find (getMsg idMsg) todos of
-      Just todo -> do
-        modifyTVar t (filter (/=todo))
-        pure $ MessageId idMsg -- OK ✅
-      Nothing   -> pure $ MessageId 0 -- did nothing
+  todos <- readTVarIO t
+  case find (getMsg idMsg) todos of
+    Just todo -> do
+      atomically $ modifyTVar t (filter (/=todo))
```

Comment:
Now I see this is wrong! `t` could change in between calls, so everything must be in the same `atomically` block.